src/System.Private.CoreLib/shared/System/Decimal.cs

```diff
@@ -553,13 +553,13 @@ internal static void GetBytes(in decimal d, byte[] buffer)
             buffer[15] = (byte)(d.flags >> 24);
         }
 
-        internal static decimal ToDecimal(byte[] buffer)
+        internal static decimal ToDecimal(ReadOnlySpan<byte> span)
         {
-            Debug.Assert((buffer != null && buffer.Length >= 16), "[ToDecimal]buffer != null && buffer.Length >= 16");
-            int lo = ((int)buffer[0]) | ((int)buffer[1] << 8) | ((int)buffer[2] << 16) | ((int)buffer[3] << 24);
-            int mid = ((int)buffer[4]) | ((int)buffer[5] << 8) | ((int)buffer[6] << 16) | ((int)buffer[7] << 24);
-            int hi = ((int)buffer[8]) | ((int)buffer[9] << 8) | ((int)buffer[10] << 16) | ((int)buffer[11] << 24);
-            int flags = ((int)buffer[12]) | ((int)buffer[13] << 8) | ((int)buffer[14] << 16) | ((int)buffer[15] << 24);
+            Debug.Assert((span != null && span.Length >= 16), "[ToDecimal]buffer != null && buffer.Length >= 16");
```

Comment:
The `span != null` should be removed: spans can't be null.

The assert string should also replace "buffer" with "span" and remove the corresponding "!= null" portion.

Reply:
Fixed.